sample-ios-app/src/iosCommonMain/kotlin/com.badoo.reaktive.sample.ios/Timer.kt

```diff
@@ -0,0 +1,26 @@
+import com.badoo.reaktive.observable.*
+import com.badoo.reaktive.scheduler.computationScheduler
+import com.badoo.reaktive.scheduler.mainScheduler
+import com.badoo.reaktive.single.Single
+import platform.Foundation.NSOperationQueue
+
+fun calculateFibonacci(): Single<List<Pair<Int, Int>>> =
+    observableOf(37, 38, 39, 40, 41)
+        .subscribeOn(computationScheduler)
+        .doOnBeforeNext { println("Should be background thread: ${!isMainThread()}") }
+        .map {
+            // some off-thread computations
+            Pair(it, fibonacci(it))
+        }
+        .observeOn(mainScheduler)
+        .doOnBeforeNext { println("Should be main thread: ${isMainThread()}") }
+        .toList()
+
+private fun fibonacci(n: Int): Int =
+    when (n) {
+        0 -> 0
+        1 -> 1
+        else -> fibonacci(n - 1) + fibonacci(n - 2)
```

Comment:
C'mon, use dynamic programming :-D

Reply:
I don't want to use dynamic programming, I want this calculation be slow and executed on background thread. Replaced with `sleep`.